src/services/smart-fields-values-injector.js

```diff
@@ -68,16 +77,22 @@ function SmartFieldsValuesInjector(record, modelName, fieldsPerModel, depth = 0)
 
   this.perform = () =>
     P.each(schema.fields, (field) => {
-      if (!record[field.field]) {
+      if (record
+        && record.dataValues
+        && !Object.prototype.hasOwnProperty.call(record.dataValues, field.field)) {
```

Comment:
Why not doing this: 
```suggestion
        && !record.dataValues.hasOwnProperty(field.field)) {
```

Reply:
Which I think is much cleaner

Reply:
I wanted to do that in the first place in the original PR, but the linter does not let me :/

Reply:
what about `field.field in record.dataValues` ?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in

Reply:
Ok, the explanation is here: https://eslint.org/docs/rules/no-prototype-builtins

> In ECMAScript 5.1, Object.create was added, which enables the creation of objects with a specified [[Prototype]]. Object.create(null) is a common pattern used to create objects that will be used as a Map. This can lead to errors when it is assumed that objects will have properties from Object.prototype. This rule prevents calling some Object.prototype methods directly from an object.
>
> Additionally, objects can have properties that shadow the builtins on Object.prototype, potentially causing unintended behavior or denial-of-service security vulnerabilities. For example, it would be unsafe for a webserver to parse JSON input from a client and call hasOwnProperty directly on the resulting object, because a malicious client could send a JSON value like {"hasOwnProperty": 1} and cause the server to crash.
>
> To avoid subtle bugs like this, it's better to always call these methods from Object.prototype. For example, foo.hasOwnProperty("bar") should be replaced with Object.prototype.hasOwnProperty.call(foo, "bar").

I learned something

Reply:
Unfortunately using `!field.field in record.dataValues` won't work and will break the test :/

From the doc:
**in operator**
```
The in operator returns true if the specified property is in the specified object or its prototype chain.
...
The in operator returns true for properties in the prototype chain. (If you want to check for only non-inherited properties, use Object.prototype.hasOwnProperty() instead.)
```

**hasOwnProperty**
```
The hasOwnProperty() method returns a boolean indicating whether the object has the specified property as its own property (as opposed to inheriting it).
```